lib/wprof/gen_prof.rb

```diff
@@ -0,0 +1,62 @@
+class GenProf
+  def initialize(event, rec_type)
+    @event = event
+    @db_runtime = WProf::Config.get_value(:db_runtime)
+    @async_report = WProf::Config.get_value(:async)
+    generate_profiling(rec_type)
+  end
+
+  def generate_profiling(rec_type)
+    common_params
+    case rec_type
+    when :standard
+      app_params
+    when :service
+      service_params
+    when :custom
+      custom_params
+    end
+    @async_report ? WprofReporter.perform_async(@params, rec_type) : WprofReporter.new.perform(@params, rec_type) # rubocop:disable Metrics/LineLength
```

Comment:
As mention above, move it to its own method and feel free to use `if/else` statement 😉

Reply:
Done.